core/unit_test/CMakeLists.txt

```diff
@@ -305,11 +305,16 @@ if(Kokkos_ENABLE_CUDA)
       UnitTestMain.cpp
       cuda/TestCuda_InterOp_Init.cpp
   )
+
+  # Run with arguments to test both CMake
+  # and kokkos command line parsing
   KOKKOS_ADD_EXECUTABLE_AND_TEST(
     UnitTest_CudaInterOpStreams
     SOURCES
       UnitTestMain.cpp
       cuda/TestCuda_InterOp_Streams.cpp
+    ARGS
+      --kokkos-num-devices 1
```

Comment:
Did you mean to check that the argument was indeed passed on the C++ side?

Reply:
No, I didn't intend to go that far

Reply:
I am hesitant to call this a test but sure it is better that what I initially proposed :)

Reply:
@crtrorr suggested we actually use this and test it in the initialization unit tests